inflection/src/inflection/util/TypeConversionUtils.cpp

```diff
@@ -45,15 +39,13 @@ TypeConversionUtils::convert(const ::std::exception& e, UErrorCode* status)
         } else if (dynamic_cast<const ::inflection::exception::IndexOutOfBoundsException*>(&e) != nullptr) {
             *status = U_INDEX_OUTOFBOUNDS_ERROR;
         } else if (dynamic_cast<const ::inflection::exception::InvalidConfigurationException*>(&e) != nullptr) {
-            *status = U_INVALID_FORMAT_ERROR;
+            *status = U_INVALID_STATE_ERROR;
         } else if (dynamic_cast<const ::inflection::exception::IOException*>(&e) != nullptr) {
             *status = U_INVALID_FORMAT_ERROR;
         } else if (dynamic_cast<const ::inflection::exception::MissingResourceException*>(&e) != nullptr) {
             *status = U_MISSING_RESOURCE_ERROR;
         } else if (dynamic_cast<const ::inflection::exception::NullPointerException*>(&e) != nullptr) {
             *status = U_MEMORY_ALLOCATION_ERROR;
```

Comment:
A direct correlation of this error does not exist in ICU. There are a lot of scenarios where null is provided, and it's not allowed. I recommend that ICU adds such a value to UErrorCode. The meaning of this error is compatible with Java's NullPointerException.